server/SwipeDecision.py

```diff
@@ -84,7 +84,29 @@ def showProfile(shownUserId):
                 gender = row[4]
                 workPlace = row[5]
                 profile = row[6]
-                
+            
+            #uses set intersection to compare matches of each user to get a mutual friends list
+            currentUserMatchSet = getMatchIds(currentUserId, cursor, connection)
+            shownUserMatchSet = getMatchIds(shownUserId, cursor, connection)
+            mutualFriendIdList = currentUserMatchSet.intersection(shownUserMatchSet)
+
+            # if the user does not have a any mutual friends, don't run the name query, and return the request
+            if len(mutualFriendIdList) == 0:
+                connection.commit()
+                cursor.close()
+                return {"response": "Success", 
+                "firstName": firstName, 
+                "interests": interests, 
+                "description": description, 
+                "age": age, 
+                "gender": gender,
+                "workPlace": workPlace,
+                "profilePicture": profile,
+                "mutualFriendAmount": 0,
```

Comment:
This has changes from the mutual friends branch, I think you should merge that into development first